### PR TITLE
Update BenchmarkDotNet

### PR DESCRIPTION
**Pull request by BrennanConroy**
Trying out the newest release

### CODE REVIEW

File: NuGet.config [hunk 1]
```diff
@@ -14,8 +14,6 @@
     <add key="dotnet31-transport" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet3.1-transport/nuget/v3/index.json" />
     <!-- Used for the Rich Navigation indexing task -->
     <add key="richnav" value="https://pkgs.dev.azure.com/azure-public/vside/_packaging/vs-buildservices/nuget/v3/index.json" />
-    <!-- Used for BenchmarkDotNet prerelease packages -->
-    <add key="benchmark-dotnet-prerelease" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/benchmark-dotnet-prerelease/nuget/v3/index.json" />
```
Comment (dougbu) [inline NuGet.config:18]: Great to see our feed count go down❕

**Review by dougbu — APPROVED** [review]
Hope this works. If the test failure(s) deserves a quarantine, please do 😀

**Comment by BrennanConroy** [comment]
https://github.com/dotnet/aspnetcore/pull/32908

**Pull request merged**